### PR TITLE
Fix error handling in date2str()

### PR DESCRIPTION
**Pull request by alejandro-colomar**

### CODE REVIEW

File: lib/date_to_str.c [hunk 1]
```diff
@@ -54,6 +33,8 @@ date_to_str(size_t size, char buf[size], long date)
 		return;
 	}
 
-	(void) strftime(buf, size, "%Y-%m-%d", tm);
-	buf[size - 1] = '\0';
+	if (strftime(buf, size, "%Y-%m-%d", tm) == 0) {
+		(void) strtcpy(buf, "future", size);
```
Comment (hallyn) [inline lib/date_to_str.c:37]: This looks fine - but of course the return isn't needed, and then the braces aren't needed.  Up to you whether you want to change that or want me to just merge it as is.

**Comment by alejandro-colomar** [comment]
On Wed, Nov 22, 2023 at 12:59:21PM -0800, Serge Hallyn wrote:
> @hallyn commented on this pull request.
> 
> 
> 
> > @@ -54,6 +33,8 @@ date_to_str(size_t size, char buf[size], long date)
>  		return;
>  	}
>  
> -	(void) strftime(buf, size, "%Y-%m-%d", tm);
> -	buf[size - 1] = '\0';
> +	if (strftime(buf, size, "%Y-%m-%d", tm) == 0) {
> +		(void) strtcpy(buf, "future", size);
> 
> This looks fine - but of course the return isn't needed, and then the
> braces aren't needed.  Up to you whether you want to change that or
> want me to just merge it as is.

Heh, I didn't think of the return; it was inertia from the ifs above.
I'll update it.  Thanks!

-- 
<https://www.alejandro-colomar.es/>

**Comment by alejandro-colomar** [comment]
v2 changes:

-  Remove superfluous return, and then the braces [@hallyn ]

```
$ git range-diff shadow/master gh/date2str date2str 
1:  2322bf89 ! 1:  cf6a2406 lib/date_to_str.c: strftime(3) leaves the buffer undefined on failure
    @@ Commit message
         been removed when I changed it to use strlcpy(3); however, I didn't
         check we didn't need it anymore.
     
    +    Cc: Serge Hallyn <serge@hallyn.com>
         Signed-off-by: Alejandro Colomar <alx@kernel.org>
     
      ## lib/date_to_str.c ##
    @@ lib/date_to_str.c: date_to_str(size_t size, char buf[size], long date)
      
     -  (void) strftime(buf, size, "%Y-%m-%d", tm);
     -  buf[size - 1] = '\0';
    -+  if (strftime(buf, size, "%Y-%m-%d", tm) == 0) {
    ++  if (strftime(buf, size, "%Y-%m-%d", tm) == 0)
     +          (void) strtcpy(buf, "future", size);
    -+          return;
    -+  }
      }
```

File: lib/date_to_str.c (hunk 1 above)
Comment (alejandro-colomar) [reply lib/date_to_str.c:37]: Fixed.

**Review by hallyn — APPROVED** [review]

**Comment by hallyn** [comment]
Thanks.

**Comment by alejandro-colomar** [comment]
You're welcome!  :-)

**Pull request merged**